EIPS/eip-7873.md

```diff
@@ -155,21 +155,39 @@ At the start of the block in which [EIP-3540](./eip-3540.md) activates, set the
 
 ```solidity
 {
-    /// Takes [tx_initcode_hash][salt][init_data] as input,
-    /// creates contract and returns the address or failure otherwise
+    /// Takes [tx_initcode_hash][salt][unsafe_flag][init_data] as input,
+    /// creates contract and returns the address or failure otherwise.
 
-    // init_data.length can be 0, but the first 2 words are mandatory
-    let size := calldatasize()
-    if lt(size, 64) { revert(0, 0) }
+    /// tx_initcode_hash and salt are 32 bytes wide, unsafe_flag is 1 byte wide, init_data can be any width.
+    /// If unsafe_flag is 0x00, `CALLER` is included in the salt and impacts the target address of the created account.
+    /// If unsafe_flag is non-zero, `CALLER` is excluded. Deployment by any sender will result in same target address. 
+    /// Deployments using non-zero unsafe_flag are susceptible to front-running.
 
-    calldatacopy(0, 0, size)  // copy tx_initcode_hash, salt and init_data to memory to hash
-    let final_salt := keccak256(0, size)
+    // init_data.length can be 0, but the first 65 bytes are mandatory
+    let size := calldatasize()
+    if lt(size, 65) { revert(0, 0) }
+
+    // copy tx_initcode_hash, salt, unsafe_flag and init_data to memory to hash.
+    // note that the unsafe_flag is included to prevent pre-image attacks on the two paths to calculate salt below.
+    calldatacopy(0, 0, size)
```

Comment:
FWIW, I believe this implements @shemnon's suggestion.

```suggestion
    // revert if tx_initcode_hash not provided.
    if lt(size, 32) {
        revert(0, 0)
    }

    // copy parameters to memory to hash.
    // note that the unsafe_flag is included to prevent pre-image attacks on the two paths to calculate salt below.
    calldatacopy(0, 0, size)

    // increase size if salt or unsafe_flag not provided, defaulting to zero.
    if lt(size, 65) {
        size := 65
    }
```

Reply:
I had it drafted similar, but did the "zero-padding-defaulting" on the `calldatacopy` step, i.e. I expand `size` one step earlier. More expensive but maybe safer.

Anyway, by "moot" you mean to keep it as is (no zero padding by default, require 65 bytes)? I think simpler is better.

Reply:
Yes, let's just require 65 bytes. This comment should be resolved.

Reply:
agree this seems too complex without much benefit, I would require all arguments